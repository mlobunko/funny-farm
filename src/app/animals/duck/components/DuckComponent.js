import React from 'react';
import PropTypes from 'prop-types';

import { StyledDuckComponent } from './styles';

export const DuckComponent = ({ isPlaying, playSound }) => (
  <StyledDuckComponent onClick={playSound} chat={isPlaying} />
);

export default DuckComponent;
DuckComponent.displayName = 'DuckComponent';

DuckComponent.propTypes = {
  isPlaying: PropTypes.bool.isRequired,
  playSound: PropTypes.func.isRequired
};
