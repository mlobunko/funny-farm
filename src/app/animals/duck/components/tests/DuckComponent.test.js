import React from 'react';
import { shallow } from 'enzyme';

import { DuckComponent } from '../DuckComponent';

const isPlaying = true;
const playSound = jest.fn();

describe('render', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <DuckComponent isPlaying={isPlaying} playSound={playSound} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});

describe('props', () => {
  it('playSound onClick', () => {
    const wrapper = shallow(
      <DuckComponent isPlaying={isPlaying} playSound={playSound} />
    );
    expect(wrapper.prop('onClick')).toEqual(playSound);
  });
});
