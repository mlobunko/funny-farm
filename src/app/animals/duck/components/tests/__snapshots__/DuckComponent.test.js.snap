// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders correctly 1`] = `
<StyledDuckComponent
  chat={true}
  onClick={[Function]}
/>
`;
